feat(tarefa-5): adiciona o gamestore-typeorm-api com o diretório tarefa-5

ThiagoOMiranda opened the pull request:
Resolução da tarefa-5, desenvolver API de tema livre: Entity, Repository, Controller e Module com persistência utilizando SQLite + TypeORM. Implementar os métodos de um CRUD: Create, Read (GetById, GetAll), Update, Delete.

Complemento da atividade: Criar os DTOs considerando o modelo escolhido e avaliar e escolher uma solução entre Mapper e Adapter para a conversão entre Entities e DTOs.

ThiagoOMiranda committed:
feat(tarefa-4): adiciona o product-api com o diretório tarefa-4

ThiagoOMiranda committed:
Fix(tarefa-4): correções na estrutra de arquivos

ThiagoOMiranda committed:
Fix(tarefa-4): correções nos arquivos para entrega e inclusão do arquivo README

ThiagoOMiranda committed:
feat(tarefa-5): adiciona o gamestore-typeorm-api com o diretório tarefa-5

ThiagoOMiranda committed:
feat(tarefa-5): Complemento da atividade 5 - Validações de DTOs

ThiagoOMiranda committed:
fix(tarefa-5) Remove os arquivos da tarefa-4 que subiram no commit da tarefa-5.

LucasPereiraCL commented:
Excelente trabalho na implementação, Thiago, parabéns! Como uma dica para otimizar ainda mais as próximas revisões, procure incluir apenas os arquivos essenciais que você de fato alterou, como os controllers, services e DTOs. Evitar arquivos de configuração geral, como o app.module, ajuda a manter o foco do revisor diretamente na sua lógica e torna o processo mais ágil para todos. Continue com o ótimo trabalho e essa atenção aos detalhes!

ThiagoOMiranda committed:
fix(tarefa-5) Substitui o projeto com persistência em SQLite para a implementação em TypeORM com banco de dados local.
feat(tarefa-5) Complementa a atividade com a criação e implementação de ParsePipe Customizado.

ThiagoOMiranda commented:
O complemento da atividade 5 foi adicionad e o projeto foi atualizado para TypeORM com banco de dados local no lugar do SQLite implementado anteriormente. Obs: Nessa versão do projeto o app.module.ts só foi incluido ao repositório remoto para evidenciar a configuração do banco de dados e propriedades aplicadas.
